airbyte-integrations/connectors/source-facebook-marketing/pyproject.toml

```diff
@@ -17,7 +17,7 @@ include = "source_facebook_marketing"
 
 [tool.poetry.dependencies]
 python = "^3.9,<3.12"
-airbyte-cdk = "^2.3.0"
+airbyte-cdk = "^3.5.0"
```

Comment:
this is a pretty big bump. Let's make sure to test with live-tests

Reply:
Live tests are ongoing...
```bash
airbyte-ci connectors --name=source-facebook-marketing build --architecture=linux/arm64

poetry run pytest src/live_tests/regression_tests --connector-image=airbyte/source-facebook-marketing --pr-url=https://github.com/airbytehq/airbyte/pull/41039 --connection-id=81d16659-48bf-44f3-ad98-17f4561e225e
```

Will let you know results

Reply:
Changing connection id as the other was not available for live tests (EU)

```
poetry run pytest src/live_tests/regression_tests --connector-image=airbyte/source-facebook-marketing --pr-url=https://github.com/airbytehq/airbyte/pull/41958 --connection-id=8d7631c6-edd4-4fe6-86ac-7d7efa3261f3
```

Reply:
[Connection with errors link](https://cloud.airbyte.com/workspaces/6705a0ab-302f-4175-813f-3f539241750a/connections/8d7631c6-edd4-4fe6-86ac-7d7efa3261f3/job-history)

Note: there is a problem currently with [requests in live tests](https://airbytehq-team.slack.com/archives/C06GNN6BERK/p1720799281300279?thread_ts=1720647416.715299&cid=C06GNN6BERK) that are not reflected in the marketing

![image](https://github.com/user-attachments/assets/853e8357-00c5-473a-a82d-15cef70ed416)

![image](https://github.com/user-attachments/assets/bed60027-ae1d-4a3e-9ed2-d6cf4527f221)

![image](https://github.com/user-attachments/assets/c99d60cd-0ece-4712-b0eb-d941ec4fa09d)

I can see the logging we added in the target logs on my local:
<img width="258" alt="image" src="https://github.com/user-attachments/assets/521ca4f4-d24b-4ce9-8197-05e9185ba91d">

```json
{"level": "WARN", "message": "Stream ads_insights_action_type: the following fields are deprecated and cannot be synced. {'conversion_lead_rate', 'cost_per_conversion_lead'}. Refresh the connection's source schema to resolve this warning.", "stack_trace": null}

{"level": "WARN", "message": "Stream ads_insights: the following fields are deprecated and cannot be synced. {'conversion_lead_rate', 'cost_per_conversion_lead'}. Refresh the connection's source schema to resolve this warning.", "stack_trace": null}

```

Reply:
I did more testing on a different connection. It was not reported to have problems but wanted to check another case to grow confidence in this solution cc @girarda 

[Connection](https://cloud.airbyte.com/workspaces/072f637e-ad0e-4e14-81e1-c5f58462e177/connections/24a2d9fd-bc1a-4606-8807-41ef9e75a3cd)

![image](https://github.com/user-attachments/assets/c484a7ce-6fd7-4788-8dbd-823a63fdb717)

![image](https://github.com/user-attachments/assets/ce56e79a-e298-4eb0-8900-688dac69542c)

![image](https://github.com/user-attachments/assets/71f0bd97-2701-4176-b8cd-f27f1e49eaae)

Also no errors found in airbyte messages of target:
<img width="1466" alt="image" src="https://github.com/user-attachments/assets/6a6fd83c-45bf-4bc8-b4cb-757fb4811372">

**Note: I couldn't find our warning for "the following fields are deprecated..." but this would be expected if there were no problems with the configured catalog; I just wanted to check that a healthy running connection would continue.**